processor/groupbyattrsprocessor/processor_test.go

```diff
@@ -691,3 +687,30 @@ func retrieveMetric(metrics pdata.MetricSlice, name string, metricType pdata.Met
 	}
 	return pdata.Metric{}, false
 }
+
+func someResourceSpans(attrs pdata.AttributeMap, count int) pdata.Traces {
+	traces := pdata.NewTraces()
+	for i := 0; i < count; i++ {
+		ils := traces.ResourceSpans().AppendEmpty().InstrumentationLibrarySpans().AppendEmpty()
+		span := ils.Spans().AppendEmpty()
+		span.SetName(fmt.Sprint("foo-", i))
+		attrs.CopyTo(span.Attributes())
+	}
+	return traces
+}
+
+func TestCompactingTraces(t *testing.T) {
+	spans := someResourceSpans(attrMap, 100)
+
+	gap := createGroupByAttrsProcessor(zap.NewNop(), []string{})
+
+	processedSpans, err := gap.processTraces(context.Background(), spans)
+	assert.NoError(t, err)
+
+	assert.Equal(t, 100, spans.ResourceSpans().Len())
+	assert.Equal(t, 1, processedSpans.ResourceSpans().Len())
+	rss := processedSpans.ResourceSpans().At(0)
+	assert.Equal(t, 0, rss.Resource().Attributes().Len())
+	// All spans are now under same resourcespans
+	assert.Equal(t, 100, rss.InstrumentationLibrarySpans().At(0).Spans().Len())
```

Comment:
Can you add some assertions to determine what happens with the instrumentation library, if it's not specified/tested elsewhere?

Reply:
Sure thing, that was just a quick PoC, addressing the comments right now